src/main/java/net/coderbot/iris/gl/shader/StandardMacros.java

```diff
@@ -0,0 +1,202 @@
+package net.coderbot.iris.gl.shader;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.lwjgl.opengl.GL11;
+import org.lwjgl.opengl.GL11C;
+import org.lwjgl.opengl.GL20;
+import org.lwjgl.opengl.GL20C;
+
+import net.minecraft.SharedConstants;
+import net.minecraft.util.Util;
+
+public class StandardMacros {
+
+	private static final Pattern SEMVER_PATTERN = Pattern.compile("(?<major>\\d+)\\.(?<minor>\\d+)\\.*(?<bugfix>\\d*)(.*)");
+
+
+	/**
+	 * Gets the current mc version String in a 5 digit format
+	 *
+	 * @return mc version string
+	 * @see <a href="https://github.com/sp614x/optifine/blob/9c6a5b5326558ccc57c6490b66b3be3b2dc8cbef/OptiFineDoc/doc/shaders.txt#L696-L699">Optifine Doc</a>
+	 */
+	public static String getMcVersion() {
+		String version = SharedConstants.getGameVersion().getReleaseTarget(); //release target so snapshots are set to the higher version
```

Comment:
Style: comments on a separate line, with a space between the `//` and the comment text

Also clarified the implication of the comment

```suggestion
		// release target so snapshots are set to the higher version
		//
		// For example, 21w10a would return 1.17
		String version = SharedConstants.getGameVersion().getReleaseTarget();
```